app/com/criteo/drelephant/heuristics/GarmadonTransferHeuristic.java

```diff
@@ -42,20 +42,42 @@
             + HEURISTIC_RESULT_TABLENAME
             + " WHERE yarn_app_result_id = :yarn_app_result_id";
 
+    private static final int MAX_ROW = 10000;
+
     public static void transfer() {
+        int nbRow = 0;
+
         // Select all app in garmadon table
         try {
             LOGGER.debug("Select all distinct ready yarn_app_result_id from {}", HEURISTIC_RESULT_TABLENAME);
             List<SqlRow> rows = Ebean.createSqlQuery(SELECT_ALL_DISTINCT_APP_SQL).findList();
 
-            for (SqlRow row : rows) {
-                String yarn_app_result_id = row.getString("yarn_app_result_id");
-                AppResult appResult = Ebean.find(AppResult.class, yarn_app_result_id);
-                if (appResult != null) {
-                    LOGGER.info("Insert data for {}", yarn_app_result_id);
-                    insertData(appResult, yarn_app_result_id);
-                } else {
-                    treatResultNotInDrElephantDb(yarn_app_result_id);
+            Transaction transaction = Ebean.beginTransaction();
+
+            try {
+                for (SqlRow row : rows) {
+                    nbRow++;
+                    String yarn_app_result_id = row.getString("yarn_app_result_id");
+                    AppResult appResult = Ebean.find(AppResult.class, yarn_app_result_id);
+                    if (appResult != null) {
+                        LOGGER.info("Insert data for {}", yarn_app_result_id);
+                        insertData(appResult, yarn_app_result_id);
+                    } else {
+                        treatResultNotInDrElephantDb(yarn_app_result_id);
+                    }
+                    if (nbRow >= MAX_ROW) {
+                        nbRow = 0;
+                        transaction.commit();
+                        transaction = Ebean.beginTransaction();
+                    }
+                }
+                transaction.commit();
+            } catch (Exception e) {
+                LOGGER.error("Unexpected error occurred while deleting useless garmadon heuristics", e);
+                throw new RuntimeException(e);
+            } finally {
+                if (transaction != null) {
```

Comment:
This will always be true, no?

Reply:
update the code to end trasaction
what happen -> nothing